// Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
syntax = "proto3";

package Schola;

import "Spaces.proto";

message AgentDefinition {
    string name = 1;
    DictSpace obs_space = 2;
    DictSpace action_space = 4;
    bool normalize_obs = 6;
    bool normalize_actions = 7;
}

message EnvironmentDefinition
{
    map<int32, AgentDefinition> agent_definitions = 1;
}

message TrainingDefinition
{   
    repeated EnvironmentDefinition environment_definitions = 1;
}
